### DISCUSSION TITLE
Feature Request; Hide Points

### DISCUSSION THREAD

**Discussion by chrisl8** [Freika/dawarich#364] [Ideas]
We can delete points now 🥳  but what if we could **hide** and **unhide** them!

My idea is adding a field in the `points` table for `hidden` that is 0 by default and set it to 1 (unless Postgress has bools?) to indicate a point is hidden. Then you could hide points by setting them to 1, and eventually the user interface could:
1. Be set to by default only work with non-hidden points in all things (stats, views, etc.)
2. Points could be hidden in in the map interface just like they can be deleted now, but also **un**-hidden (see next).
3. On the map interface you could turn hidden points on/off or even only show hidden points, for comparing what was hidden over time and such, and of course for un-hiding previously hidden points.

**Comment by Freika** [comment]
Assuming we're having hundreds of thousands points, what would the process of hiding look like? That sounds like a very tedious task, but I'm genuinely interested

**Reply by chrisl8** [reply]
Two methods:
1. Literally the same as the delete option, just now we could "un-delete" them. Also it would be nice to be able to select like a month or year and only show "hidden" points to find patterns.
2. Via external tools/scripts/etc. that users (like me) might make that process the data and hide points based on varying criteria. I already have some scripts to delete points, but I've realized I wish I could un-delete them and/or look at them later to see what those patterns look like.

**Comment by chrisl8** [comment]
As an example, here is what some of my normal data looks like:
![image](https://github.com/user-attachments/assets/7b466ec0-2926-48d1-8b34-ffe2a2d508e4)
(I'll post a cleaned up version of this data later as a comparison.)

It is quite messy. I will clean up some by hand but at this point I also have a script that uses the data to clean up some points.

Up until now I've been just deleting points I know are "unreliable", but I decided what I'd like to do is just "hide" those points and then I'd like to be able to come back and look at all "hidden" points over a day, a week, a month, a year and see if there is a pattern to what is happening.

I'd also like to be able to easily recover points that I hid later if I messed up.

What I'm doing now is writing out points to another table before deleting them, so I can at least recover them and see them in a table, but I don't have a way to visualize them. My first thought was that if rather than delete points I could just update a field in the table to "hide" them, then it would be easy to visualize them.

**Reply by Freika** [reply]
Got it, thanks for the explanation. Can't see why it shouldn't be a feature, so I'll implement it, hopefully soon
